src/vanta_ledger/routes/documents.py

```
@@ -0,0 +1,171 @@
+            "message": "Document uploaded and processed successfully",
+            "doc_id": result['doc_id'],
+            "original_filename": result['original_filename'],
+            "type": result['analysis']['type'],
+            "summary": result['analysis']['summary'],
+            "security": result["security"]
+        }
+
+    except HTTPException:
+        raise
+    except Exception as e:
+        raise HTTPException(status_code=500, detail=f"Document processing failed: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)